### DISCUSSION TITLE
Add Toasts to OMAP

### DISCUSSION THREAD

**Discussion by JulianFeinauer** [pragmaticindustries/open-machinery-plattform#24] [Ideas]
I had a short discussion with @PaulMirsch this morning and we came to the topic "Toasts", i.e. "Notifications" that are shown after successfull (or failed?) operations.
I like the idea very much and would go for a "simple" implementation following our general development pattern.
My initial idea was to add a stimulus controller that takes a set of divs as "toasts" (similar to the map controller that takes a div for each point) and takes care about the rendering (styled via tailwind ofc) and also hiding / removing them after some time.

@mario-glitch what are your thoughts about this?

**Comment by JulianFeinauer** [comment]
Short Addition: One advantage of this approach is, that the toasts and their contents can be set via the backend and will simply be displayed when showing the page. So all logic stays in the backend and no async stuff is required (like websockets or other compexity).
Depending on the level of complexity we could either
- add it direct to template variables in the view
or
- have a context processor for that and views can simply push notifications for certain users there that will then be displayed whenever the respective user loads a page

**Comment by JulianFeinauer** [comment]
Short update here... there already is a working stimulus controller here: https://www.stimulus-components.com/docs/stimulus-notification/ and it works perfectly fine to change the css in a way that multiple toasts can be displayed:
![grafik](https://user-images.githubusercontent.com/17098762/164009409-b4f71ee6-7546-4181-8423-8e50a57d5bda.png)

**Reply by NiklasMerz** [reply]
I'm a fan of not doing this ourselves. Creating it ourselves can be harder than expected (thinking about the modal controller).

We used the bootstrap toast in the HMI because it was really easy to use.
